src/tls13.c

```diff
@@ -3383,7 +3383,7 @@ int DoTls13ServerHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
 
         /* Update counts to reflect change of message type. */
         ssl->msgsReceived.got_hello_retry_request = 1;
-        ssl->msgsReceived.got_server_hello--;
+        ssl->msgsReceived.got_server_hello = 0;
```

Comment:
I can remove this section now, since it is done below... Changing and will test  / push update.